File: src/app/models/Refrigerateur.ts
export interface Refrigerateur {
    id?: number;                
    placesDisponibles?: number;  // Number of available places (in KG)
    imageUrl?: string;           // URL for the image of the refrigerator
    creerpar?: string;
    donateur?: {                 // Minimal User object for the donateur
      iduser?: number;   
      login?:string;        // User ID of the donateur
      nom?: string;             // Optional: Name of the donateur
      prenom?: string;          // Optional: First name of the donateur
      numtel?: string;

    };
  }
  

File: src/main/java/com/food/Entities/Refrigerateur.java
package com.food.Entities;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.io.Serializable;
import java.util.Date;
@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Refrigerateur implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Float placesDisponibles; // Number of available places to store food (in KG)

    private String imageUrl; // Assuming the image URL is stored in the DB

    @ManyToOne
    private User donateur;

}